test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs

```diff
@@ -593,5 +593,118 @@ public void TestCrossValidationMacroWithStratification()
                 }
             }
         }
+
+        [Fact]
+        public void TestCrossValidationMacroWithNonDefaultNames()
+        {
+            string dataPath = GetDataPath(@"adult.tiny.with-schema.txt");
+            using (var env = new TlcEnvironment(42))
+            {
+                var subGraph = env.CreateExperiment();
+
+                var textToKey = new ML.Transforms.TextToKeyConverter();
+                textToKey.Column = new[] { new ML.Transforms.TermTransformColumn() { Name = "Label1", Source = "Label" } };
+                var textToKeyOutput = subGraph.Add(textToKey);
+
+                var hash = new ML.Transforms.HashConverter();
+                hash.Column = new[] { new ML.Transforms.HashJoinTransformColumn() { Name = "GroupId1", Source = "Workclass" } };
+                hash.Data = textToKeyOutput.OutputData;
+                var hashOutput = subGraph.Add(hash);
+
+                var learnerInput = new Trainers.FastTreeRanker
```

Comment:
>FastTreeRanker [](start = 48, length = 14)

You can add Transforms.RandomNumberGenerator to emulate Weight column.

Reply:
I couldn't add it to this test because the ranker trainer expects the weights to be constant per group. I added it instead to another CV macro test.

---
In reply to: [192858690](https://github.com/dotnet/machinelearning/pull/291#discussion_r192858690) [](ancestors = 192858690)